### PR TITLE
chore: document client methods

### PR DESCRIPTION
**Pull request by veu**
## Summary

Document the client methods and reorganize the client type to create readable documentation.

## Description

* Put all client methods directly into `ContentfulClientApi` to generate readable documentation. Small caveats:
  * `Locales` type parameter is now also available for methods that don’t need it.
  * Client properties that can’t be used are now typed as `never` instead of omitted.
* Move `LocaleOption` into `…Queries` types to improve readability.

## Motivation and Context

TypeDoc doesn’t do well with complex types. We can spend all day coming up with clever dynamic solutions for types but at least for those types that we want to cover with generated documentation we have to be a little bit more straightforward.

### CODE REVIEW

File: test/types/queries/entry-queries.test-d.ts
Comment (Cyberxon) [inline test/types/queries/entry-queries.test-d.ts:1]: why did you add the `undefined` value in tests here?

File: lib/create-contentful-api.ts [hunk 1]
```diff
@@ -139,17 +137,12 @@ export default function createContentfulApi<OptionType extends ChainOptions>(
     })
   }
 
-  async function getEntry<EntrySkeleton extends EntrySkeletonType = EntrySkeletonType>(
-    id: string,
-    query: EntryQueries & LocaleOption = {}
-  ) {
-    return makeGetEntry<EntrySkeleton>(id, query, options)
+  async function getEntry(id, query = {}) {
```
[the remark this answers is not in the document]
Comment (veu) [reply lib/create-contentful-api.ts:140]: It’s internal code that’s not visible to users. The effort to type it properly (after adding the generic to `EntryQueries` is too much for something that won’t improve the experience.

File: lib/types/client.ts [hunk 2]
```diff
@@ -1,18 +1,35 @@
...
 import { ContentType, ContentTypeCollection } from './content-type'
 import { Space } from './space'
 import { LocaleCode, LocaleCollection } from './locale'
-import { AssetQueries, EntriesQueries, EntrySkeletonType, TagQueries } from './query'
+import { AssetsQueries, EntriesQueries, EntrySkeletonType, TagQueries } from './query'
 import { SyncCollection, SyncQuery } from './sync'
 import { Tag, TagCollection } from './tag'
 import { AssetKey } from './asset-key'
-import { EntryQueries, LocaleOption } from './query/query'
+import { AssetQueries, EntryQueries } from './query/query'
```
[the remark this answers is not in the document]
Comment (veu) [reply lib/types/client.ts]: done

**Review by Cyberxon — APPROVED** [review]
LGTM, thank you for the clarifications!
